drivers/bluetooth/hci/slz_hci.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+
+#include <sl_btctrl_linklayer.h>
+#include <sl_hci_common_transport.h>
+#include <pa_conversions_efr32.h>
+#include <sl_bt_ll_zephyr.h>
+
+#define SL_BT_CONFIG_ACCEPT_LIST_SIZE	1
+#define SL_BT_CONFIG_MAX_CONNECTIONS	1
+#define SL_BT_CONFIG_USER_ADVERTISERS	1
+#define SL_BT_CONTROLLER_BUFFER_MEMORY  CONFIG_BT_SILABS_HCI_BUFFER_MEMORY
+#define SL_BT_SILABS_LL_STACK_SIZE	1024
+
+static K_KERNEL_STACK_DEFINE(slz_ll_stack, SL_BT_SILABS_LL_STACK_SIZE);
+static struct k_thread slz_ll_thread;
+
+void rail_isr_installer(void)
+{
+	IRQ_CONNECT(RDMAILBOX_IRQn, 0, RDMAILBOX_IRQHandler, NULL, 0);
+	IRQ_CONNECT(RAC_SEQ_IRQn, 0, RAC_SEQ_IRQHandler, NULL, 0);
+	IRQ_CONNECT(RAC_RSM_IRQn, 0, RAC_RSM_IRQHandler, NULL, 0);
+	IRQ_CONNECT(PROTIMER_IRQn, 0, PROTIMER_IRQHandler, NULL, 0);
+	IRQ_CONNECT(MODEM_IRQn, 0, MODEM_IRQHandler, NULL, 0);
+	IRQ_CONNECT(FRC_IRQn, 0, FRC_IRQHandler, NULL, 0);
+	IRQ_CONNECT(BUFC_IRQn, 0, BUFC_IRQHandler, NULL, 0);
+	IRQ_CONNECT(AGC_IRQn, 0, AGC_IRQHandler, NULL, 0);
+}
+
+/**
+ * @brief Transmit HCI message using the currently used transport layer.
+ * The HCI calls this function to transmit a full HCI message.
+ * @param[in] data Data buffer to be transmitted.
+ * @param[in] len Length of the data.
+ * @return 0 - on success, or non-zero on failure.
+ */
+uint32_t hci_common_transport_transmit(uint8_t *data, int16_t len)
+{
+	struct net_buf *buf;
+
+	switch (data[0]) {
+	case h4_event:
+		uint8_t flags, evt = data[1];
+
+		buf = bt_buf_get_evt(evt, false, K_FOREVER);
+		net_buf_add_mem(buf, data + 1, len - 1);
+		flags = bt_hci_evt_get_flags(evt);
+
+		if (flags & BT_HCI_EVT_FLAG_RECV_PRIO) {
+			bt_recv_prio(buf);
+		} else {
+			bt_recv(buf);
+		}
+
+		break;
+	case h4_acl:
+		buf = bt_buf_get_rx(BT_BUF_ACL_IN, K_FOREVER);
+		net_buf_add_mem(buf, data + 1, len - 1);
+		bt_recv(buf);
+		break;
+	default:
+		break;
+
+	}
+
+	sl_btctrl_hci_transmit_complete(0);
+	return 0;
+}
+
+static int slz_bt_send(struct net_buf *buf)
+{
+	int rv = 0;
+
+	switch (bt_buf_get_type(buf)) {
+	case BT_BUF_ACL_OUT:
+		net_buf_push_u8(buf, h4_acl);
+		break;
+	case BT_BUF_CMD:
+		net_buf_push_u8(buf, h4_command);
+		break;
+	default:
+		rv = -EINVAL;
+		goto done;
+	}
+
+	rv = hci_common_transport_receive(buf->data, buf->len, true);
+	if (!rv) {
+		goto done;
+	}
+
+done:
+	net_buf_unref(buf);
+	return rv;
+}
+
+static int slz_bt_open(void)
+{
+	/* Start RX thread */
+	k_thread_create(&slz_ll_thread, slz_ll_stack,
+			K_KERNEL_STACK_SIZEOF(slz_ll_stack),
+			(k_thread_entry_t)slz_ll_thread_func, NULL, NULL, NULL,
+			K_PRIO_COOP(CONFIG_BT_DRIVER_RX_HIGH_PRIO), 0,
+			K_NO_WAIT);
+
+	rail_isr_installer();
+
+	sl_rail_util_pa_init();
+
+	sl_btctrl_init_mem(SL_BT_CONTROLLER_BUFFER_MEMORY);
```

Comment:
What if this returns zero? According to the public documentation, this seems possible.